### PR TITLE
JoltC: bindings wip

### PR DESCRIPTION
**Pull request by hazeycode**
Declares:
- JPH_AABox
- JPH_Plane
- JPH_Sphere
- JPH_MotionProperties

Defines bindings for
- JPH::MotionProperties

### CODE REVIEW

File: libs/zjolt/libs/JoltC/JoltC.h [hunk 1]
```diff
@@ -151,13 +167,76 @@ typedef struct JPH_BroadPhaseLayerInterfaceVTable JPH_BroadPhaseLayerInterfaceVT
...
+
+    // 3rd cache line (least freqently used)
+    // 4 byte aligned
+    JPH_Sphere          sleep_test_spheres[3];
+    float               sleep_test_timer;
+
+#ifdef JPH_ENABLE_ASSERTS
+    JPH_MotionType      cached_motion_type;
+#endif
+};
```
Comment (hazeycode) [inline libs/zjolt/libs/JoltC/JoltC.h:238]: I'm having some trouble to get the size of the struct to be correct. Any ideas @michal-z?

Comment (hazeycode) [reply libs/zjolt/libs/JoltC/JoltC.h:245]: > I'm having some trouble to get the size of the struct to be correct. Any ideas @michal-z?

I think this is the last thing and then this PR is ready

**Comment by hazeycode** [comment]
https://github.com/michal-z/zig-gamedev/pull/113#discussion_r973513816
i think this is the last thing before this PR is ready

**Comment by michal-z** [comment]
Thanks, I will look at the `JPH_MotionProperties` size later today.

**Comment by michal-z** [comment]
@hazeycode You missed one field :)
```
float                   mMaxAngularVelocity;
```

**Comment by michal-z** [comment]
Actually, fields that are `Float3` can't be 4 element arrays, sorry. So, `force` and `torque` must have 3 elements.

**Pull request merged**